### PR TITLE
Enable Scalafix On The Meta Build

### PR DESCRIPTION
**Pull request by isomarcte**
We have quite a bit of code to support code generation, and we are going to get more in the upcoming PRs. It would be nice to have the same quality and linting checks we have on the normal code.

@armanbilge I was going to use sbt-typelevel-settings for this, but I need to disable `-Ywarn-unused-import` and because sbt-typelevel-settings sets scalacOptions in the `projectSettings` this becomes difficult. Is there any reason we don't set the scalacOptions in the `globalSettings` in sbt-typelevel-settings? My understanding was that the `globalSettings` was the normal place for default values and it makes it simple to override/modify at the project/build level. But I may be off base here.

### CODE REVIEW

**Comment by armanbilge** [comment]
> @armanbilge I was going to use sbt-typelevel-settings for this, but I need to disable `-Ywarn-unused-import` and because sbt-typelevel-settings sets scalacOptions in the `projectSettings` this becomes difficult.

Sorry, how does this make it difficult? Isn't there only one project in the build?

**Comment by isomarcte** [comment]
Oh, you're right, I was being silly and had `ThisBuild / scalacOptions -= "-Ywarn-unused-import"` instead of just `scalacOptions -= "-Ywarn-unused-import"`. 

I'll change that. Unless I'm mistaken, I do think it still can make customization difficult for normal multi-project builds though. Like if I have 10 projects and I want to remove one compiler flag globally, I have to do that at a per-project level, right, rather than once in the build scope? 

In any case, I'll move this discussion over to sbt-typelevel, since it's not relevant here.

**Comment by armanbilge** [comment]
Yes I do agree about that, thanks for raising the issue. We are currently overhauling scalac-options for sbt-typelevel 0.5.x so we can do that as well. See https://github.com/typelevel/sbt-typelevel/issues/256#issuecomment-1248431036.

File: project/Versions.scala [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (armanbilge) [inline project/Versions.scala:1]: Is this an empty file?

Comment (isomarcte) [reply project/Versions.scala]: Gone

**Review by armanbilge — APPROVED** [review]

**Pull request merged**